File: Semestre_1/TP_C++/pre vacances/tp4/tp4/exo1.cpp
#include<iostream>
using namespace std;

int maxi(int x, int y) {
	if (x < y) {
		return y;
	}
	else {
		return x;
	}
}

int moyenne(int x, int y) {
	float moy = 0.0;
	moy = (x + y) / 2;
	return moy;
}

int main() {
	int x, y;
	float moy = 0.0;
	cout << "entrer 2 entiers\n";
	cin >> x >> y;
	moy = moyenne(x, y);
	cout << moy;
	return 0;
}

File: Semestre_2/TP_Java/qualite du dev/tp1/exo1/src/exo1/Exo1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package exo1;

import java.util.Scanner;

/**
 *
 * @author p2103455
 */
public class Exo1 {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        
        Scanner scan;
        scan = new Scanner(System.in);
        
        System.out.println("entrer des entiers et terminé votre liste par 'fin': ");
        
        String valeur = new String();
        int moyenne = 0;
        int nb_valeur = 0;
        while (true){
            valeur = scan.next();
            if("fin".equals(valeur)){
                break;
            }
            try {
                moyenne =( moyenne + Integer.parseInt(valeur));
                nb_valeur++;
            }   
            catch (NumberFormatException e) {
                System.out.println("la valeur numero " + (nb_valeur+1) + " n'est pas un int");
            }
        }
        moyenne = moyenne/ nb_valeur;
        System.out.println(moyenne);
        
    }
    
}
